### PR TITLE
Clear compaction code

### PR DESCRIPTION
**Pull request by zhanglingzhe0820**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/TsFileManagement.java [hunk 1]
```diff
@@ -66,325 +62,464 @@
...
+      logger.error(e.getMessage(), e);
+    } finally {
+      seqFile.writeUnlock();
     }
   }
 
-  public void mergeEndAction(
-      List<TsFileResource> seqFiles, List<TsFileResource> unseqFiles, File mergeLog) {
-    logger.info("{} a merge task is ending...", storageGroupName);
+  public static int getMergeLevel(File file) {
```
Comment (EJTTianYu) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/TsFileManagement.java:418]: The current IoTDB data file format is as follows：timestamp-version-level-unseq merge cnt, is this the required merge level?

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/TsFileManagement.java [hunk 2]
```diff
@@ -66,325 +62,464 @@
...
+  private List<SortedSet<TsFileResource>> newSequenceTsFileResources(Long k) {
+    List<SortedSet<TsFileResource>> newSequenceTsFileResources = new ArrayList<>();
+    for (int i = 0; i < seqLevelNum; i++) {
+      newSequenceTsFileResources.add(
+          new TreeSet<>(
+              (o1, o2) -> {
+                try {
+                  int rangeCompare =
+                      Long.compare(
+                          Long.parseLong(o1.getTsFile().getParentFile().getName()),
```
Comment (EJTTianYu) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/TsFileManagement.java:435]: I doubt the rangeCompare of the parentFile，as the current file dir is like sgName/timepartition/virtualSG

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java [hunk 3]
```diff
@@ -533,9 +520,9 @@ private void recoverLevelCompaction() {
         CompactionTaskManager.getInstance()
             .submitTask(
                 logicalStorageGroupName,
-                tsFileManagement.new LevelCompactionRecoverTask(this::closeCompactionMergeCallBack));
+                compactionScheduler.new CompactionRecoverTask(this::closeCompactionMergeCallBack));
```
Comment (EJTTianYu) [inline server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java:523]: as level compaction has been renamed to inner space compaction, the function name should be renamed

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/TsFileManagement.java [hunk 4]
```diff
@@ -66,325 +62,464 @@
...
+      Math.max(IoTDBDescriptor.getInstance().getConfig().getUnseqLevelNum(), 1);
 
   public TsFileManagement(String storageGroupName, String storageGroupDir) {
     this.storageGroupName = storageGroupName;
     this.storageGroupDir = storageGroupDir;
   }
 
-  public void setFullMerge(boolean forceFullMerge) {
-    isForceFullMerge = forceFullMerge;
+  public Pair<List<List<TsFileResource>>, List<List<TsFileResource>>> forkCurrentFileList(
```
Comment (liutaohua) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/TsFileManagement.java:83]: I think we should wrap a `class` here, 

```
class LevelCompactionFiles{
    List<List<tsFileSource>> getSeqResource(), 
    List<List<tsFileSource>> getUnseqResource(), 
    
    List<TsFileResource> getSeqResource(int level), 
    List<TsFileResource> getUnseqResource(int level)
}
```

**Pull request merged**